pkg/storage/replica_command.go

```diff
@@ -1971,8 +1871,12 @@ func (r *Replica) sendSnapshot(
 			},
 		},
 		RangeSize: r.GetMVCCStats().Total(),
-		// Recipients can choose to decline preemptive snapshots.
-		CanDecline: snapType == SnapshotRequest_PREEMPTIVE,
+		// Recipients currently cannot choose to decline any snapshots.
+		// TODO(ajwerner): Should a recipient be able to decline a LEARNER snapshot?
+		// It seems like yes. The recipient may know that it does not want to get
+		// added to a range. Confer with danhhz/tbg/darin about the implications of
+		// making this true for SnapshotRequest_LEARNER.
```

Comment:
This was just discussed on https://github.com/cockroachdb/cockroach/pull/44247#issuecomment-577668089

TLDR: when you refuse the learner, you've already added it to the desc, so we need to be careful. We don't want to roll back replication changes en masse which is what would happen here.